ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -315,35 +227,40 @@ S3FileOpen(WT_FILE_SYSTEM *fileSystem, WT_SESSION *session, const char *name,
         return (EINVAL);
     }
 
+    // Check if the file exists and get the size
+    bool objExists;
+    wt_off_t objSize;
+    if ((ret = S3FileExistsAndGetSize(fileSystem, session, name, &objExists, &objSize)) != 0) {
+        s3->log->LogErrorMessage("S3FileOpen: error checking if the file exists. Ret code:" + ret);
+        return (EINVAL);
+    }
+    if (!objExists) {
+        s3->log->LogErrorMessage(
+          "S3FileOpen: object named " + std::string(name) + " does not exist in the bucket.");
+        return (ENOENT);
+    }
+
+    // Check if there is already an existing file handle open.
```

Comment:
Ah, we are sharing file handles between threads - generally we do not, as file handles contain the seek pointer. But I suppose it is okay in the cloud cases since file handles do not contain any state saved between the read requests.